### ISSUE TITLE
tinyMCE failing SRI integrity check

### ISSUE THREAD

**Issue by elliot-sawyer** [Firesphere/silverstripe-csp-headers#17]
When accessing the CMS, I get the following error:

`[Warning] file_get_contents(/path/to/mysite//assets/_tinymce/tinymce-cms-afd8c8731c.js): failed to open stream: No such file or directory`

The cause seems to be in Model/SRI.php::onBeforeWrite():
```
       // Since this is called from CSP Backend, an SRI for external files is automatically created
        if (!Director::is_site_url($this->File)) {
            /** @var Client $client */
            $client = Injector::inst()->get(Client::class);
            $result = $client->request('GET', $this->File);
            $body = $result->getBody()->getContents();
        } else {
            $body = file_get_contents(Director::baseFolder() . '/' . $this->File);
        }
        $hash = hash(CSPBackend::SHA384, $body, true);

        $this->SRI = base64_encode($hash);
```

If the URL is considered to be a site URL, it derives its path from the `baseFolder()`. This isn't always the case, if the CMS is dumping TinyMCE into another location.

I can raise a PR that adds an additional check for site files that might exist in the `publicFolder()`, which allows the hash to be calculated correctly

**Comment by robingram** [comment]
@Firesphere Are there any plans to tag a release containing this? I can target `master` for now though.

**Comment by Firesphere** [comment]
Hmmm, there's no tag for the merge?

I'll fix that up tomorrow, I thought I tagged it, but then again, I've got some stuff going on, so I may have missed it.

**Comment by Firesphere** [comment]
@robingram Quick check says that it should be included in 1.0.1

**Comment by robingram** [comment]
I don’t think so. We were on 1.0.1. That was released on 5 April but the PR was merged on 14 May

**Comment by Firesphere** [comment]
Interesting.
Fixed with releasing 1.0.2
https://github.com/Firesphere/silverstripe-csp-headers/releases/tag/1.0.2
